Source/WebCore/html/ImageData.cpp

```diff
@@ -36,9 +36,24 @@
 
 namespace WebCore {
 
-static CheckedUint32 computeDataSize(const IntSize& size)
+static PixelFormat toPixelFormat(ImageDataStorageFormat storageFormat)
 {
-    return PixelBuffer::computeBufferSize(PixelFormat::RGBA8, size);
+    switch (storageFormat) {
+    case ImageDataStorageFormat::Uint8:
+        break;
+    case ImageDataStorageFormat::Float16:
+#if HAVE(HDR_SUPPORT)
+        return PixelFormat::RGBA16F;
+#else
+        break;
+#endif
```

Comment:
I'd love `PixelFormat::RGBA16F` to be unconditional! 😄 But maybe that would be hard to implement on some non-Cocoa platforms?
I did make `ImageDataStorageFormat::Float16` unconditional because I think:
- The proposal suggested that it should be available everywhere.
- It's possible to implement it without backed support, with `put/getImageData` doing conversions on the fly.
- IIRC I initially made it conditional at the IDL level, and got compilation errors in generated code, and didn't pursue it further.

But we could decide to keep both in sync if that's preferable. To be discussed...